learning_observer/learning_observer/lms_integration.py

```diff
@@ -0,0 +1,287 @@
+import json
+import recordclass
+import string
+import aiohttp
+import aiohttp.web
+
+import learning_observer
+import learning_observer.runtime
+import learning_observer.google
+import learning_observer.constants as constants
+import learning_observer.settings as settings
+import pmss
+
+pmss.register_field(
+    name="lms_api",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The Canvas Base API URL",
+    required=True
+)
+
+cache = None
+
+class Endpoint(recordclass.make_dataclass("Endpoint", ["name", "remote_url", "doc", "cleaners", "lms"], defaults=["", None])):
+    def arguments(self):
+        return extract_parameters_from_format_string(self.remote_url)
+
+    def _local_url(self):
```

Comment:
Again, docstring on all functions. `pycodestyle` and `pylint` will flag that.

Reply:
Added docstring here